### ISSUE TITLE
Add Date/Time to Packages' Changelog on Pub Dev

When viewing a Package's Changelog on pub.dev it would be great to see the date/time when the version was published.

### ISSUE THREAD

**Issue by grahamsmith** [dart-lang/pub-dev#5867]
![FVy55LzVUAAmzpV.jpeg](https://user-images.githubusercontent.com/1571527/174952215-bd06d73c-f701-496a-84d5-2089889f96db.jpeg)

Ref original tweet about this: 
https://twitter.com/whoisgraham/status/1539309744812216325?s=20&t=IG_e3iX0U3zbQNHIbiIFMQ

**Comment by grahamsmith** [comment]
@mit-mit tagging you in per Tim's instructions 👍

**Comment by sigurdm** [comment]
The changelog is presented directly from the actual uploaded package - currently there is nothing actually correlating it with the published versions existing on pub.dev.

There is the "versions" tab that presents the actual versions and the date they got uploaded.

We have talked about aligning these two into one when possible. There are a number of open questions, such as:

How do we ensure that all changelogs are parsing well and contain the right versions?

Which versions changelog should be used to show on the version tab? (latest / latest stable / latest stable plus the entry from each prerelease after that).
 
I think we want to do this, but it is not on our immediate roadmap.

**Comment by sigurdm** [comment]
![image](https://user-images.githubusercontent.com/8613953/176150877-4830dc63-48dd-40e6-b991-1a696ff68fe4.png)

If you click on the "x days ago" you get to see the actual date.

**Comment by grahamsmith** [comment]
Thanks for getting back to me.

I think this is a UX problem more than some complicated coding issue.

It's good to know that I can click to reveal the date.

I think it shows we need a bit of work as there's no indication that I can click to reveal.

Imagine there's bigger fish to fry but also this can't be a long hard issue to solve either.

**Comment by sigurdm** [comment]
> I think it shows we need a bit of work as there's no indication that I can click to reveal.

Yeah - that's a good point, the discoverability is not great - I wonder how we could make this more obvious.

@isoos do you have good ideas? Perhaps it should show the date on hover-instead of click, like here on github:
![image](https://user-images.githubusercontent.com/8613953/176171630-9bfb79ba-8294-468a-ba5f-903dcb1ea624.png)

**Comment by grahamsmith** [comment]
Cant we just print under the X days ago in a smaller font? Would be good for screen readers etc.

**Comment by isoos** [comment]
> Perhaps it should show the date on hover-instead of click, like here on github:

We do both: if you hover over the "X ago..." string, it also shows the date with the browser's default hover-popup.

> Cant we just print under the X days ago in a smaller font?

We use this at multiple places, and on some occasions this may work, but on others this will feel very strange, e.g. on the scores tab we have "We analyzed this package __4 days ago__, and awarded...".

For screen readers, we could probably add the `aria-label` attribute (or more if needed), and fill it with the same content as with the regular `title` attribute. Would that be enough?

**Comment by sigurdm** [comment]
> We do both: if you hover over the "X ago..." string, it also shows the date with the browser's default hover-popup.

Sorry - did not see that at first

> For screen readers, we could probably add the aria-label attribute (or more if needed), and fill it with the same content as with the regular title attribute. Would that be enough?

I would guess that screen readers already can use the title tag... Not sure if we can do much more here...

Maybe ... just maybe - the dotted line under "x days ago" that appears when hovering could be there all the time to indicate that it is an interactive element.

![image](https://user-images.githubusercontent.com/8613953/176202762-74ba6cb2-c676-42b1-87cb-1f72e4f81356.png)
